src/app/registration/registration.directive.ts

```diff
@@ -0,0 +1,22 @@
+import {Directive, Input, HostListener} from '@angular/core';
+
+
+@Directive({
+    selector: '[validateOnBlur]',
```

Comment:
This should be `appValidateOnBlur` according to the [style guide: style-02-08](https://angular.io/guide/styleguide#style-02-08).